### PR TITLE
Climaxing/masturbation changes, exposing through clothing, breastmilk and masturbation

### PR DESCRIPTION
**Pull request by ktccd**
[Changelogs]: # (Please make a changelog if you're adding, removing or changing content that'll affect players. This includes, but is not limited to, new features, sprites, sounds; balance changes; map edits and important fixes)

:cl: ktccd
add: Breasts can now be used to masturbate/milk yourselves.
add: You can now expose genitals through clothing using a verb in the IC tab.
tweak: Pressing the Arousal HUD symbol will now bring up several options to use. Pull or be pulled by someone who's exposed to have them appear as an option.
tweak: Made hexacamphor/hexacrocin less verbose and trigger forced climaxes less often to avoid incredible spam.
tweak: Being forced to orgasm while pulled or pulling someone now counts them as your selected partner.
/:cl:

[why]: # (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:) 
Lots of things:
Exposing genitalia has been asked for a while now, and helps work around things were normal jackets would count as covering the groin.
Hexacrocin and hexacamphor were too verbose and the forced orgasms happening too often.
Breastcode existed to support it, but there was no way for players to milk themselves. This is now supported.
My previous rework of the masturbation was a bit lacking in options and the bloated look of the proc really added to that.
You can now also cum on any food item, but I'm not putting that in the changelogs.... I feel a lot of people would break the OOC-consent rule by feeding people their semen, without thinking about needing consent....

~~DO NOT MERGE JUST YET, I gotta clean up some comments I think I forgot?~~
Ok, removed the comments.

### CODE REVIEW

File: code/citadel/cit_arousal.dm [hunk 1]
```diff
@@ -181,224 +181,365 @@
...
+							"<span class='green'>You feel yourself about to orgasm.</span>", \
+							"<span class='green'>You feel yourself about to orgasm.</span>")
+		if(do_after(src, mb_time, target = src))
+			if(total_fluids > 5)
+				fluid_source.reaction(src.loc, TOUCH, 1, 0)
+			fluid_source.clear_reagents()
+			src.visible_message("<span class='danger'>[src] orgasms[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""], using [p_their()] [G]!</span>", \
+								"<span class='green'>You climax[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""] with your [G].</span>", \
+								"<span class='green'>You climax using your [G].</span>")
+		if(G.can_climax)
```
Comment (deathride58) [inline code/citadel/cit_arousal.dm]: You forgot a tab here. Without that tab, your arousal will reset even if the timer gets interrupted.

Comment (ktccd) [reply code/citadel/cit_arousal.dm]: ah, dang tabs! Good eye ^^.

Comment (ktccd) [reply code/citadel/cit_arousal.dm]: Fixed it.

File: code/citadel/cit_arousal.dm [hunk 2]
```diff
@@ -181,224 +181,365 @@
...
+								"<span class='green'>You orgasm with [L][spillage ? ", spilling out of them":""], using your [G].</span>", \
+								"<span class='green'>You have climaxed with someone[spillage ? ", spilling out of them":""], using your [G].</span>")
+	else //knots and other non-spilling orgasms
+		if(do_after(src, mb_time, target = src) && in_range(src, L))
+			fluid_source.trans_to(L, total_fluids)
+			total_fluids = 0
+			src.visible_message("<span class='danger'>[src] climaxes with [L], [p_their()] [G] spilling nothing!</span>", \
+								"<span class='green'>You ejaculate with [L], your [G] spilling nothing.</span>", \
+								"<span class='green'>You have climaxed inside someone, your [G] spilling nothing.</span>")
+	if(G.can_climax)
```
Comment (deathride58) [inline code/citadel/cit_arousal.dm]: bad tabs 2: electric boogaloo
if you get separated from your partner or you move, your arousal will reset, but you will not cum. slightly realistic, but probably not the intended effect.

Comment (ktccd) [reply code/citadel/cit_arousal.dm]: Fixed. I added the check to the spillage also. I think my idea was for the command to run after either the spillage and non-spillage versions, but I forgot about the do_after part being important for both.

**Comment by ktccd** [comment]
Got some feedback I'm going to implement, don't merge this just yet.

File: code/citadel/cit_reagents.dm [hunk 3]
```diff
@@ -153,14 +153,13 @@
 	if(prob(33))
 		if(M.getArousalLoss() >= 100 && ishuman(M) && M.has_dna())
 			var/mob/living/carbon/human/H = M
-			to_chat(H, "<span class='love'>Your libido is going haywire!</span>")
-			H.mob_climax(forced_climax=TRUE)
+			if(prob(50)) //Less spam
```
Comment (TalkingCactus) [inline code/citadel/cit_reagents.dm:156]: I'd go 10 here.

Comment (ktccd) [reply code/citadel/cit_reagents.dm:156]: I tested that and it almost never triggered.
It's prob(33) THEN prob(50), resulting in about a prob(16) total. If I made it prob(10), the end result would be a 3% chance to trigger :S.

Comment (TalkingCactus) [reply code/citadel/cit_reagents.dm:156]: It doesn't need to tell you that often, really. >20u of the stuff lasts a while anyway, so you'll still get the message a fair few times while it metabolizes.

Comment (ktccd) [reply code/citadel/cit_reagents.dm:156]: yeah... but 3% is extremely low. I just remember testing that and I basically never saw a message or was forced to climax...

**Comment by Iamgoofball** [comment]
Call a priest

**Comment by ktccd** [comment]
If you were really so horrified by this, you wouldn't be scouring our PR's to find something to comment on ^^.
Tell us the truth, Goof, you're either secretly playing on our server on an alt or you find a perverse joy in being horrified ^_^.

**Comment by Iamgoofball** [comment]
i put all the downstreams that auto-mirror PRs on notifications so I can make sure important bugfixes get merged

this unfortunately comes with the job

**Comment by Iamgoofball** [comment]
chrome_2017-08-18_16-50-08

**Comment by silicons** [comment]
@Iamgoofball ktccd is a hero

**Comment by Spacedong** [comment]
WHAT THE FUCK

**Pull request merged**